src/plots/gl3d/scene.js

```diff
@@ -31,6 +30,32 @@ var createAxesOptions = require('./layout/convert');
 var createSpikeOptions = require('./layout/spikes');
 var computeTickMarks = require('./layout/tick_marks');
 
+var isMobile = require('is-mobile');
+var tablet = isTablet();
+
+function isTablet() {
+    var navigator = window.navigator;
+    var userAgent = navigator.userAgent;
```

Comment:
and similarly here, if `is-mobile` guards against undefined `navigator.userAgent`, we should too here.

Reply:
Good call.

Reply:
Revised in https://github.com/plotly/plotly.js/pull/4546/commits/436fca46dad2226913b07f3ab0d01b04cb0ad03c.